src/analysis/specific_limits.lean

```diff
@@ -218,41 +249,77 @@ end
 
 section edist_le_geometric
 
-variables [emetric_space α] (r C : nnreal) (hr : r < 1) {f : ℕ → α}
+variables [emetric_space α] (r C : ennreal) (hr : r < 1) (hC : C ≠ ⊤) {f : ℕ → α}
   (hu : ∀n, edist (f n) (f (n+1)) ≤ C * r^n)
 
-include hr hu
+include hr hC hu
 
-/-- If `edist (f n) (f (n+1))` is bounded by `C * r^n`, `r < 1`, then `f` is a Cauchy sequence.-/
+/-- If `edist (f n) (f (n+1))` is bounded by `C * r^n`, `C ≠ ∞`, `r < 1`,
+then `f` is a Cauchy sequence.-/
 lemma cauchy_seq_of_edist_le_geometric : cauchy_seq f :=
 begin
-  norm_cast at hu,
-  exact cauchy_seq_of_edist_le_of_summable _ hu
-   (summable_mul_left C $ summable_geometric_nnreal hr)
+  refine cauchy_seq_of_edist_le_of_tsum_ne_top _ hu _,
+  rw [ennreal.mul_tsum, ennreal.tsum_geometric],
+  refine ennreal.mul_ne_top hC (ennreal.inv_ne_top.2 _),
+  exact ne_of_gt (ennreal.zero_lt_sub_iff_lt.2 hr)
 end
 
-/-- If `edist (f n) (f (n+1))` is bounded by `C * r^n`, `r < 1`, then the distance from
+omit hr hC
+
+/-- If `edist (f n) (f (n+1))` is bounded by `C * r^n`, then the distance from
 `f n` to the limit of `f` is bounded above by `C * r^n / (1 - r)`. -/
 lemma edist_le_of_edist_le_geometric_of_tendsto {a : α} (ha : tendsto f at_top (𝓝 a)) (n : ℕ) :
   edist (f n) a ≤ (C * r^n) / (1 - r) :=
 begin
-  norm_cast at hu,
-  rw [← ennreal.coe_one], rw_mod_cast [← ennreal.coe_div (ne_of_gt $ nnreal.sub_pos.2 hr)],
-  convert edist_le_tsum_of_edist_le_of_tendsto _ hu
-    (summable_mul_left C $ summable_geometric_nnreal hr) ha _,
-  refine eq.symm (tsum_eq_has_sum _),
-  simp only [pow_add, nnreal.div_def, mul_assoc],
-  exact has_sum_mul_left C (has_sum_mul_left (r ^ n) (has_sum_geometric_nnreal hr))
+  convert edist_le_tsum_of_edist_le_of_tendsto _ hu ha _,
+  simp only [pow_add, ennreal.mul_tsum, ennreal.tsum_geometric, ennreal.div_def, mul_assoc]
 end
 
-/-- If `edist (f n) (f (n+1))` is bounded by `C * r^n`, `r < 1`, then the distance from
+/-- If `edist (f n) (f (n+1))` is bounded by `C * r^n`, then the distance from
 `f 0` to the limit of `f` is bounded above by `C / (1 - r)`. -/
 lemma edist_le_of_edist_le_geometric_of_tendsto₀ {a : α} (ha : tendsto f at_top (𝓝 a)) :
   edist (f 0) a ≤ C / (1 - r) :=
-by simpa only [pow_zero, mul_one] using edist_le_of_edist_le_geometric_of_tendsto r C hr hu ha 0
+by simpa only [pow_zero, mul_one] using edist_le_of_edist_le_geometric_of_tendsto r C hu ha 0
 
 end edist_le_geometric
 
+section edist_le_geometric_two
+
+variables [emetric_space α] (C : ennreal) (hC : C ≠ ⊤) {f : ℕ → α}
```

Comment:
Ditto: remove `hC` from the list of global variables, and only put it in the lemmas when relevant.